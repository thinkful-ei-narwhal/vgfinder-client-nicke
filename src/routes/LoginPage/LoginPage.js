import React, { Component } from "react";
import LoginForm from "../../components/LoginForm/LoginForm";
import { Section } from "../../components/Utils/Utils";
import Header from "./../../components/Header/Header";
import "./LoginPage.css";

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  };

  handleLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || "/homepage";
    history.push(destination);
  };

  render() {
    return (
      <>
        <Header pathName={this.props.location.pathname} />
        <Section>
          <h2>Login</h2>
          <LoginForm
            className="LoginPage"
            onLoginSuccess={this.handleLoginSuccess}
          />
        </Section>
      </>
    );
  }
}
