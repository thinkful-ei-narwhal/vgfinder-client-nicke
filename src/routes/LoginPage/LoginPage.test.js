import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginPage from "./LoginPage";

describe(`LoginPage component`, () => {
  it("renders a LoginPage component without props", () => {
    const wrapper = shallow(<LoginPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
