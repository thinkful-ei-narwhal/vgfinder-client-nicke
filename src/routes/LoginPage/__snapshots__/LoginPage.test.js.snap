// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders a LoginPage component without props 1`] = `
<Fragment>
  <Header />
  <Section>
    <h2>
      Login
    </h2>
    <LoginForm
      className="LoginPage"
      onLoginSuccess={[Function]}
    />
  </Section>
</Fragment>
`;
